syntax = "proto3";
package rofi.messages;

import "connectorPacket.proto";
import "connectorPowerState.proto";

message ConnectorCmd {
    enum Type {
        NO_CMD = 0;
        GET_STATE = 1;
        CONNECT = 2;
        DISCONNECT = 3;
        CONNECTOR_EVENT = 4;
        PACKET = 5;
        CONNECT_POWER = 6;
        DISCONNECT_POWER = 7;
        POWER_STATE = 8;
    }

    int32 connector = 1;
    Type cmdType = 2;

    Packet packet = 3;
    PowerState.Line line = 4;
}
